core/src/io/anuke/mindustry/world/blocks/power/ItemLiquidGenerator.java

```diff
@@ -92,7 +92,12 @@ public boolean shouldConsume(Tile tile){
     }
 
     @Override
-    public void update(Tile tile){
+    public void update(Tile tile, float amountOfPowerUsed){
+    	// Should not update or do anything if we havn't consumed any power. This will leave the tower in a 'pause' state rather than consuming materials.
+    	if (amountOfPowerUsed <= 0f)
+    	{
+    		return;
```

Comment:
What if a generator is sending out 50 power, but the grid requests 25? It will continue to burn fuel at the same rate as before, e.g. at 100% combustion speed. This will work only in situations where there is **no** power being requested from the grid, and those situations are very rare and don't present much of a problem in real games. The real problem arises when there's a large surplus of power that isn't going anywhere, but *part of it is still being consumed*.